app/graphql/mutations/account/grant_site_permission.rb

```diff
@@ -0,0 +1,29 @@
+class Mutations::Account::GrantSitePermission < Mutations::Base
+  argument :account, ID,
+    required: true,
+    description: 'Who to grant permissions to'
+
+  argument :permission, Types::Enum::SitePermission,
+    required: true,
+    description: 'The permission to grant to this user'
+
+  field :id, ID, null: false
+  field :permissions, [Types::Enum::SitePermission], null: false
```

Comment:
The reason for the mutation response objects looking like this is to allow Apollo clients to automatically update their cache with the mutation's changes. That said, I'm not sure this even works right. I'm not super familiar with Apollo's caching.

Maybe we should instead have the mutation return the entire Account object?

Reply:
If this actually has caching benefits that is fine, if not, I would prefer the actual object (Account)